org.jacoco.core/src/org/jacoco/core/internal/BytecodeVersion.java

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal;
+
+import org.objectweb.asm.Opcodes;
+
+/**
+ * Utilities to read and modify bytecode version in bytes of class.
+ */
+public final class BytecodeVersion {
```

Comment:
ASM has an API to read/write bytecode versions, therefore we should give a reason here why we have this class. Maybe add some Javadoc: "The purpose of this class is to deal with new Java versions which are not yet supported by ASM".

Reply:
@marchof JavaDoc updated